lib/base/tlsstream.cpp

```diff
@@ -103,3 +105,65 @@ void UnbufferedAsioTlsStream::BeforeHandshake(handshake_type type)
 	}
 #endif /* SSL_CTRL_SET_TLSEXT_HOSTNAME */
 }
+
+/**
+ * Forcefully close the connection, typically (details are up to the operating system) using a TCP RST.
+ */
+void AsioTlsStream::ForceDisconnect()
+{
+	if (!lowest_layer().is_open()) {
+		// Already disconnected, nothing to do.
+		return;
```

Comment:
Is consistent with GracefulDisconnect() and shouldn't hurt – idk how Windows handles the -1 thing (https://github.com/Icinga/icinga2/pull/10005#discussion_r1844132097) if you close a closed socket.😅